### DISCUSSION TITLE
PPM-Mode: Arrangement protocol in order of switch position ?

### DISCUSSION THREAD

**Discussion by baerbar** [pascallanger/DIY-Multiprotocol-TX-Module#831] [Q&A]
Hello Everybody,
I've bought a IRangeX IRX4+ multi-module to combine my single Spectrum- and Flysky-Modules. Additionally I got a Eachine Spitfire (Volantex), which I want to stear via PPM-Port with my Graupner Hott-Transmitter, too.

The problem is that the protocol for the spitfire (V761) is located in the actual firmware 1.3.3.220 only as serial protocol on number 48, but not on the list of the switch assignment.

I studied some instruction on multi-module.org and here on Pascals github. But I'm overstrained to compile a custom _config.h in the right form. Unfortunately I don't understand the method to do this.
So, as question: is it generally possible, to re-arrange the protocols to get them on the rotary switch? First off I would be happy to get DSMX, FlySky and the V761-Protocol on the switch number 1 to 3 ...
When I understand the method, I can add some of the other protocols if needed.

Any answers are welcome. Thank you in advance!
Regards, Bernd

Used sources:
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/docs/Compiling_STM32.md
https://www.multi-module.org/using-the-module/protocol-details/v761

**Comment by pascallanger** [comment]
This is pretty easy:
1. open the _config.h file
2. by default there is 1 bank which should be enough in your case, you can increase the number of banks here : https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/139cd4c58385b48e6208b9282a2d7149846afce2/Multiprotocol/_Config.h#L428
3. modify the lines https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/139cd4c58385b48e6208b9282a2d7149846afce2/Multiprotocol/_Config.h#L434 with the protocol you want. Each line indicates a rotary switch position (first number which you do not touch) followed by the protocol (protocols and subprotocols name can be found after the tables), sub_protocol and parameters as indicated by the column headers
4. Compile and upload to the module

**Reply by baerbar** [reply]
Thank you Pascal!
If I understand you correctly, would this be sufficient?

`/*	2	*/	{PROTO_V761,	4CH	,	0	,	P_HIGH	,	NO_AUTOBIND	,	0	,	0x00000000 },`

How do I set missing parameters like "RX_Num"?

**Reply by pascallanger** [reply]
Nearly, the sub_protocol is V761_4CH as shown in the list in _Config.h: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/139cd4c58385b48e6208b9282a2d7149846afce2/Multiprotocol/_Config.h#L815
RX number is the first "0" on your line, check the column names: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/139cd4c58385b48e6208b9282a2d7149846afce2/Multiprotocol/_Config.h#L433

**Reply by baerbar** [reply]
Okay, second try:

`/*	2	*/	{PROTO_V761,	V761_4CH	,	0	,	P_HIGH	,	NO_AUTOBIND	,	0	,	0x00000000 },`


> RX number is the first "0" on your line, check the column names

I saw this, but what does it mean? Binding one or more receivers?

**Reply by pascallanger** [reply]
A RX will only recognize the TX if the RX number matches (dependant of the protocol and subprotocol).
Example: you bind RX A with RX number 1 and RX B with RX number 2. RX A will not work if you select the wrong entry containing RX number 1.

**Reply by baerbar** [reply]
Hi Pascal, I've made the test with a compiled and flashed firmware. Seems to be successful, cause I get now a working binding to a DSMX and a Flysky (old) receiver. 
That's a good base for make this configuration file better, and I'm really happy that it works!
Thank you for your fast and accurate support! You're doing a great project here.
Greetings, Bernd

**Comment by baerbar** [comment]
I'm sorry for asking agin, but as I wrote above I want to use my Graupner Hott mc-26.
Compiling the protcol order works fine now and binding with FlySky, DSM and V791 is possible.

But the channel order is not right - maybe I've have to switch to TAER, which is the default order by Graupner. Unfortunatelly I didn't found the way to do this **manually** in my **custom** files. Where do I have to look?

**Reply by baerbar** [reply]
Oh, too fast. I've to take a closer look in future. I'm found it, in _config.h

/*** TX SETTINGS ***/
/*******************/
//Modify the channel order based on your TX: AETR, TAER, RETA...
//Examples: Flysky & DEVO is AETR, JR/Spektrum radio is TAER, Multiplex is AERT...
//Default is AETR.
#define TAER

Compiled it to custom file, but it unfortunatelly didn't solve the problem with rearranged channels.

**Reply by pascallanger** [reply]
That should work. Have you checked if there is a compilation error due to a typo? May be you're using the last good compiled firmware.

**Reply by baerbar** [reply]
Compilation was without warnings and error, the date of the file was updated, too.
Nothing is changed except TAER....

1. The spitfire with the V761-prot needs a correction of throttle way on channel 1, but controls are okay.
2. Two Axion flyer with FlySky-brick needs a change of rudder and elevator (compared to my old extern FlySky-Module)
3. I tested four Spektrum-bricks from UMX models. All of them response to the binding, but then the servos go to maximum - that's it...

**Reply by baerbar** [reply]
Had checked and compiled it yesterday after a long pause again. Now it works fine, but I don't know what I made false before...
Time now for extended tests.
Thank you for helping!
